### PR TITLE
SocketAddress in AgentRegistrationData

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #2323 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > UTs pass
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/infection_monkey/network/relay/utils.py [hunk 1]
```diff
@@ -90,8 +89,7 @@ def send_remove_from_waitlist_control_message_to_relays(servers: Iterable[str]):
 
 
 def _send_remove_from_waitlist_control_message_to_relay(server: str):
-    ip, port = address_to_ip_port(server)
-    server_address = SocketAddress(IPv4Address(ip), int(port))
+    server_address = SocketAddress(*address_to_ip_port(server))
```
Comment (VakarisZ) [inline monkey/infection_monkey/network/relay/utils.py:92]: don't you need to provide kwargs to SocketAddress, since it's a pydantic model?

Comment (VakarisZ) [reply monkey/infection_monkey/network/relay/utils.py:92]: Use this once it's merged: https://github.com/guardicore/monkey/pull/2350/files#diff-69582fdb56607bef080bc859a5d7007b039dbbad57e4abae63eda498626b96b6

File: monkey/monkey_island/cc/models/agent.py [hunk 2]
```diff
@@ -26,7 +27,7 @@ class Agent(MutableInfectionMonkeyBaseModel):
     parent_id: Optional[AgentID] = Field(allow_mutation=False)
     """The ID of the parent agent that spawned this agent"""
 
-    cc_server: str = Field(default="")
+    cc_server: Optional[SocketAddress]
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/models/agent.py:30]: Optional means something or None, not something or empty string, though I doubt it's used anywhere

File: monkey/tests/unit_tests/monkey_island/cc/island_event_handlers/test_handle_agent_registration.py [hunk 3]
```diff
@@ -111,7 +112,7 @@ def test_existing_machine_updated__find_by_ip(handler, machine_repository):
         machine_hardware_id=5,
         start_time=0,
         parent_id=None,
-        cc_server="192.168.1.1:5000",
```
Comment (VakarisZ) [inline monkey/tests/unit_tests/monkey_island/cc/island_event_handlers/test_handle_agent_registration.py:114]: Extract this and build from string

File: monkey/tests/unit_tests/monkey_island/cc/models/test_agent_registration_message.py [hunk 4]
```diff
@@ -6,16 +6,18 @@
 import pytest
 
 from common import AgentRegistrationData
+from common.types import SocketAddress
 
 AGENT_ID = UUID("012e7238-7b81-4108-8c7f-0787bc3f3c10")
 PARENT_ID = UUID("0fc9afcb-1902-436b-bd5c-1ad194252484")
+SOCKET_ADDRESS = SocketAddress(ip="192.168.1.1", port="5000")
```
Comment (VakarisZ) [inline monkey/tests/unit_tests/monkey_island/cc/models/test_agent_registration_message.py:13]: re-use this in [test_handle_agent_registration.py](https://github.com/guardicore/monkey/pull/2353/files#diff-1bac01f51af89c43f7980e3e5a694e16851b967842c7863b98c739e0d4ddad96)?

**Review by VakarisZ — COMMENTED** [review]
Maybe rebase on 2323-parse-servers-as-socketaddress?

File: monkey/infection_monkey/network/relay/utils.py (hunk 1 above)
Comment (cakekoa) [inline monkey/infection_monkey/network/relay/utils.py:92]: Note: This is also modified in another PR, because the port from `address_to_ip_port` can be `None`. However, SocketAddress will validate the input because it is a pydantic model, which is also a valid solution.
This will change when the previous comment is addressed.

File: monkey/infection_monkey/network/relay/utils.py [hunk 5]
```diff
@@ -90,8 +89,7 @@ def send_remove_from_waitlist_control_message_to_relays(servers: Iterable[str]):
 
 
 def _send_remove_from_waitlist_control_message_to_relay(server: str):
```
Comment (cakekoa) [inline monkey/infection_monkey/network/relay/utils.py:91]: This should have been updated to accept the server as a `SocketAddress`. We should do this in another PR

**Review by cakekoa — APPROVED** [review]

File: monkey/infection_monkey/network/relay/utils.py (hunk 1 above)
Comment (cakekoa) [reply monkey/infection_monkey/network/relay/utils.py:92]: This will need to be addressed

**Review by cakekoa — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/models/agent.py (hunk 2 above)
Comment (shreyamalviya) [reply monkey/monkey_island/cc/models/agent.py:30]: I couldn't find any usages of cc_server where an empty string was required.

File: monkey/tests/unit_tests/monkey_island/cc/models/test_agent_registration_message.py (hunk 4 above)
Comment (shreyamalviya) [reply monkey/tests/unit_tests/monkey_island/cc/models/test_agent_registration_message.py:13]: It's just an IP address. I don't think it's worth the refactoring.

File: monkey/infection_monkey/network/relay/utils.py (hunk 5 above)
Comment (shreyamalviya) [reply monkey/infection_monkey/network/relay/utils.py:91]: Change the logic. This function doesn't exist anymore.

File: monkey/infection_monkey/monkey.py [hunk 6]
```diff
@@ -118,19 +118,15 @@ def __init__(self, args):
...
 
         server, self._island_api_client = self._connect_to_island_api()
-        # TODO: `address_to_port()` should return the port as an integer.
         self._cmd_island_ip, self._cmd_island_port = address_to_ip_port(server)
-        self._cmd_island_port = int(self._cmd_island_port)
 
-        self._island_address = SocketAddress(
-            IPv4Address(self._cmd_island_ip), self._cmd_island_port
-        )
+        self._island_address = SocketAddress(self._cmd_island_ip, self._cmd_island_port)
```
[the remark this answers is not in the document]
Comment (shreyamalviya) [reply monkey/infection_monkey/monkey.py:123]: Which PR?

File: monkey/infection_monkey/network/relay/utils.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/infection_monkey/network/relay/utils.py:92]: Which PR? Anyhow, this function has been removed now.

**Review by ilija-lazoroski — APPROVED** [review]

File: monkey/infection_monkey/monkey.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (shreyamalviya) [reply monkey/infection_monkey/monkey.py:124]: Found it. #2351
